src/main/java/levi9/UserService/service/impl/UserServiceImpl.java

```diff
@@ -59,4 +60,21 @@ public UserDto loadUserByUsername(String username) {
     }
 
 
+
+    @Override
+    public UserDto editUserDetails(String username,UserDto userDetails ) {
+            User userDetails2 = mapper.map(userDetails,User.class);
```

Comment:
You don't need to map it to User class but use that userDetails from request to read values down bellow directly (userDetails.getFirstName()...)

Reply:
solved